cli/admin/src/main/java/io/pravega/cli/admin/dataRecovery/StorageListSegmentsCommand.java

```diff
@@ -0,0 +1,167 @@
+            if (this.filePath.endsWith(Path.SEPARATOR)) {
+                this.filePath = this.filePath.substring(0, this.filePath.length()-1);
+            }
+        }
+
+        // Create a directory for storing files.
+        File dir = new File(this.filePath);
+        if (!dir.exists()) {
+            dir.mkdir();
+        }
+
+        for (int containerId = 0; containerId < this.containerCount; containerId++) {
+            File f = new File(this.filePath + Path.SEPARATOR + "Container_" + containerId + ".csv");
+            if (f.exists()) {
+                outputInfo("File '%s' already exists.", f.getAbsolutePath());
+                if (!f.delete()) {
+                    outputError("Failed to delete the file '%s'.", f.getAbsolutePath());
+                    throw new Exception("Failed to delete the file " + f.getAbsolutePath());
+                }
+            }
+            if (!f.createNewFile()) {
+                outputError("Failed to create file '%s'.", f.getAbsolutePath());
+                throw new Exception("Failed to create file " + f.getAbsolutePath());
+            }
+            this.csvWriters[containerId] = new FileWriter(f.getAbsolutePath());
+            outputInfo("Created file '%s'", f.getAbsolutePath());
+            this.csvWriters[containerId].append(String.join(",", HEADER));
+            this.csvWriters[containerId].append("\n");
+        }
+    }
+
+    @Override
+    public void execute() throws Exception {
+        outputInfo("Container Count = %d", this.containerCount);
+
+        // Get the storage using the config.
+        this.storage.initialize(CONTAINER_EPOCH);
+        outputInfo("Loaded %s Storage.", getServiceConfig().getStorageImplementation().toString());
+
+        // Gets total number of segments listed.
+        int segmentsCount = 0;
+
+        // create CSV files for listing the segments and their details
+        createCSVFiles();
+
+        outputInfo("Writing segments' details to the csv files...");
+        Iterator<SegmentProperties> segmentIterator = this.storage.listSegments();
+        while (segmentIterator.hasNext()) {
+            SegmentProperties currentSegment = segmentIterator.next();
+
+            // skip, if the segment is an attribute segment.
+            if (NameUtils.isAttributeSegment(currentSegment.getName())) {
+                continue;
+            }
+
+            segmentsCount++;
+            int containerId = this.segToConMapper.getContainerId(currentSegment.getName());
+            outputInfo(containerId + "\t" + currentSegment.isSealed() + "\t" + currentSegment.getLength() + "\t" +
+                    currentSegment.getName());
+            this.csvWriters[containerId].append(currentSegment.isSealed() + "," + currentSegment.getLength() + "," +
+                    currentSegment.getName() + "\n");
+        }
+
+        outputInfo("Closing all csv files...");
+        for (int containerId = 0; containerId < this.containerCount; containerId++) {
```

Comment:
FYI you can just do `for(FileWriter w : this.csvWriters)` instead of this loop. It's cleaner.

If you change it, do it in the `close` method too.

Reply:
Done. Thanks.